cpp/src/parquet/column_reader.h

```diff
@@ -114,8 +115,8 @@ struct CryptoContext {
   bool start_decrypt_with_dictionary_page = false;
   int16_t row_group_ordinal = -1;
   int16_t column_ordinal = -1;
-  std::shared_ptr<Decryptor> meta_decryptor;
-  std::shared_ptr<Decryptor> data_decryptor;
+  std::function<std::shared_ptr<Decryptor>()> meta_decryptor;
```

Comment:
nit: rename to `meta_decryptor_creator` or something alike for better readability.